### ISSUE TITLE
[stable/mongodb-replicaset] Doesn't start (Blocked at bootstrap)

### ISSUE THREAD

**Issue by oxyno-zeta** [helm/charts#4560]
**Is this a request for help?**:
Yes
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
Helm:
```
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.1", GitCommit:"6af75a8fd72e2aa18a2b278cfe5c7a1c5feca7f2", GitTreeState:"clean"}
```

Kubernetes:
```
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.0", GitCommit:"fc32d2f3698e36b93322a3465f63a14e9f0eaead", GitTreeState:"clean", BuildDate:"2018-03-26T16:55:54Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.5+coreos.0", GitCommit:"0d082e389e1f4311dc5d225eb77f9688c50d340a", GitTreeState:"clean", BuildDate:"2018-03-21T21:10:44Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Which chart**:
Chart:
stable/mongodb-replicaset
Version:
2.3.2

**What happened**:
Pods for the stateful states are blocked in the `bootstrap` container.
```
containers with incomplete status: [bootstrap]
```
There is no logs for this container.

**What you expected to happen**:
The stateful state must start :).

**How to reproduce it** (as minimally and precisely as possible):
Install the chart:
```sh
helm install --name mongo -f mongo-vars.yml stable/mongodb-replicaset --version 2.3.2
```
Vars:
```yml
persistentVolume:
 storageClass: local-storage
```

**Anything else we need to know**:
Cluster:
The cluster was created with [Kubespray](https://github.com/kubernetes-incubator/kubespray) with 3 machines on Scaleway.

Chart:
With no persistentVolume it is the same.

Chart 2:
When I replace installImage by another one it works !

Vars:
```yml
persistentVolume:
 storageClass: local-storage

installImage:
  name: quay.io/samsung_cnct/install-mongodb
  tag: v0.1.0
```

Tell me if your need more information.
Thanks for your help !

Oxyno-zeta

**Comment by timstoop** [comment]
Which installImage did you replace it with?

**Comment by oxyno-zeta** [comment]
Hello, I replace the default one by the Samsung_cnct.

**Comment by gfleury** [comment]
I'm having the same issue with a cluster deployed with kubespray. Basically the /work-dir/peer-finder will never exit as the statefulset is created in order and the first pod isn't ready as it is still on waiting for bootstrap. I guess that is related to the .spec.podManagementPolicy that is by default set to Ordered. Using parallel should work.

**Comment by gfleury** [comment]
Not actually, tried changing the ordered, didn't worked. Guess that is related to #1399

**Comment by gfleury** [comment]
Same issue as #1399, updating the peer-finder to the latest version it works perfect. This is why @oxyno-zeta said that changing the installImage probably to one with latest version of peer-finder worked.

**Comment by oxyno-zeta** [comment]
Hello,

Sorry for being late @gfleury .
I've updated to latest chart (3.3.1) but I continue to have this error.

I've upgraded the old one and not created a new replica set. It is the problem ?

If I upgrade to the latest chart with the Samsung install image, something is blocked... The install image found the other mongo nodes but stopping here...

If you have any ideas. 

Oxyno-zeta

**Comment by gfleury** [comment]
Hi Oxyno, 

what I meant is that on my case the cluster is NOT using the default cluster dns (cluster.local). This configuration makes peer-finder to stop working on old versions, and installImage uses an old version of peer-finder. So, for me, I've used the latest chart and I changed the installImage to gfleury/mongodb-install:latest (is just a wrapper updating the peer-finder). And it worked like a charm. My chart deployment was a fresh install as well, not an update. 
The install image code that I used is here (https://github.com/gfleury/mongodb-install).
This seems a mapped issue from #1399

**Comment by oxyno-zeta** [comment]
Thanks for your quick response @gfleury but it didn't works...

The logs:
```
My Hostname: mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron
Hosts in list: mongo-mongodb-replicaset-0.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-1.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron-cluster
2018/05/09 17:45:23 Have not found myself in list yet.
My Hostname: mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron
Hosts in list: mongo-mongodb-replicaset-0.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-1.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron-cluster
2018/05/09 17:45:24 Have not found myself in list yet.
My Hostname: mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron
Hosts in list: mongo-mongodb-replicaset-0.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-1.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron-cluster
2018/05/09 17:45:25 Have not found myself in list yet.
My Hostname: mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron
Hosts in list: mongo-mongodb-replicaset-0.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-1.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron-cluster
2018/05/09 17:45:26 Have not found myself in list yet.
My Hostname: mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron
Hosts in list: mongo-mongodb-replicaset-0.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-1.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron-cluster
2018/05/09 17:45:27 Have not found myself in list yet.
My Hostname: mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron
Hosts in list: mongo-mongodb-replicaset-0.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-1.mongo-mongodb-replicaset.default.svc.iron-cluster, mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron-cluster
```

It seems that the peer-finder find correctly the hosts but doesn't continue...

I've created my image because I don't find yours in the docker hub. (https://gitlab.com/Oxyno-zeta/mongodb-install/)

Any other idea ? :)

**Comment by burdiyan** [comment]
Having same issue. Unable to upgrade the chart from the previous release and blocked on the third init container.

**Comment by gfleury** [comment]
Hi @oxyno-zeta, 

Look the logs "mongo-mongodb-replicaset-2.mongo-mongodb-replicaset.default.svc.iron", isn't the right domain, maybe this is why, check the reason why the pod was a different name.

**Comment by oxyno-zeta** [comment]
I will probably create a new cluster later :). I think I change the default name to "iron". I can't remember sorry.

**Comment by mythmoon** [comment]
Has this problem been solved yet? Install 0.6 still has a problem.

**Comment by oxyno-zeta** [comment]
Hello,

I switch to mongodb chart. I haven't any problem with this one. I can't reproduce this bug.
Do you want to close it ?

Thanks !

Oxyno-zeta
